tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -235,27 +236,17 @@ public static void KillProcessTrees(Queue<Process> processQueue)
         /// </summary>
         /// <param name="process">The parent process</param>
         /// <returns>A list of child processes</returns>
-        /// <exception cref="NotImplementedException">Thrown if running on an OS other than Windows</exception>
+        [SupportedOSPlatform("windows")]
         public static IList<Process> GetChildProcesses(this Process process)
         {
-            // Validate platform compatibility
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                ManagementObjectSearcher processSearch = new ManagementObjectSearcher($"Select * From Win32_Process Where ParentProcessID={process.Id}");
-                IList<Process> processList = processSearch.Get()
-                    .Cast<ManagementObject>()
-#pragma warning disable CA1416 // This call can not be reached except on Windows due to the enclosing if statement
-                    .Select(mo =>
-                        Process.GetProcessById(Convert.ToInt32(mo["ProcessID"], System.Globalization.CultureInfo.InvariantCulture)))
-#pragma warning restore CA1416
-                    .ToList();
-                processSearch.Dispose();
-                return processList;
-            }
-            else
-            {
-                throw new NotImplementedException("Not implemented for this OS");
-            }
+            ManagementObjectSearcher processSearch = new ManagementObjectSearcher($"Select * From Win32_Process Where ParentProcessID={process.Id}");
+            IList<Process> processList = processSearch.Get()
+                .Cast<ManagementObject>()
+                .Select(mo =>
+                    Process.GetProcessById(Convert.ToInt32(mo["ProcessID"], System.Globalization.CultureInfo.InvariantCulture)))

```

Comment:
You might want to do that outside the loop?

Reply:
It is done in the loop because as far as I can tell grandchild processes would have the spawning child process as the ParentProcessID rather than the root parent which is why we need to search each child processes children as well, or am I missing something?